wallet/wallet.go

```diff
@@ -3438,19 +3446,19 @@ func (w *Wallet) publishTransaction(tx *wire.MsgTx) (*chainhash.Hash, error) {
 	// confirmed store. We'll avoid returning an error as the broadcast was
 	// in a sense successful.
 	//
-	// This error is returned when broadcasting/sending a transaction that
-	// has already confirmed to a btcd node.
-	case strings.Contains(err.Error(), "transaction already exists"):
+	// This error is returned when sending a transaction that has already
+	// confirmed to a btcd/bitcoind node over RPC.
+	case rpcTxConfirmed:
 		fallthrough
 
 	// This error is returned when broadcasting a transaction that has
-	// already confirmed to a bitcoind node.
-	case strings.Contains(err.Error(), "txn-already-known"):
+	// already confirmed to a btcd node over the P2P network.
+	case strings.Contains(err.Error(), "transaction already exists"):
 		fallthrough
 
-	// This error is returned when sending a transaction that has already
-	// confirmed to a bitcoind node over RPC.
-	case strings.Contains(err.Error(), "transaction already in block chain"):
+	// This error is returned when broadcasting a transaction that has
+	// already confirmed to a bitcoind node over the P2P network.
+	case strings.Contains(err.Error(), "txn-already-known"):
```

Comment:
We'll need to maintain the old _and_ new error messages in order to support the prior version as we do currently.

Reply:
This is already the case for both backends with this commit. `rpcTxConfirmed` will be caught by old (< 0.18.0) and new (>= 0.18.0) `bitcoind` versions and new `btcd` versions. `transaction already exists` will be caught by old `btcd` nodes (those before https://github.com/btcsuite/btcd/pull/1428).